server/api/eventinfo/eventinfo.model.js

```diff
@@ -60,14 +60,40 @@ EventInfoSchema.pre("save",function(next) {
   next();
 });
 
+EventInfoSchema
+  .set('toJSON', {
+      virtuals: true
+  });
+
+EventInfoSchema
+  .set('toObject', {
+      virtuals: true
+  });
 
 // VIRTUALS
 EventInfoSchema
     .virtual("isHappeningNow")
     .get(function(){
         var now = new Date();
-        // TODO iterate through times, check if event is happening
-        return true;
+        var isHappeningNow = false;
+        if(this.times.length > 0) {
+          this.times.forEach((time)=>{
+            if(time.start.getTime() < now.getTime() && time.end.getTime() > now.getTime()) {
+              isHappeningNow = true;
+            }
+          });
+        }
+        return isHappeningNow;
+    });
```

Comment:
This is fine, but I think a more functional way might look like this...

```javascript
return this.times.some((time) => time.start.getTime() < now.getTime() && time.end.getTime() > now.getTime())
```

Also is a lot shorter ;)